ray-operator/controllers/ray/common/pod.go

```diff
@@ -753,10 +754,16 @@ func generateRayStartCommand(ctx context.Context, nodeType rayv1.RayNodeType, ra
 	if _, ok := rayStartParams["num-gpus"]; !ok {
 		// Scan for resource keys ending with "gpu" like "nvidia.com/gpu".
 		for resourceKey, resource := range resource.Limits {
-			if strings.HasSuffix(string(resourceKey), "gpu") && !resource.IsZero() {
+			resourceKeyString := string(resourceKey)
+			if strings.HasSuffix(resourceKeyString, "gpu") && !resource.IsZero() {
 				rayStartParams["num-gpus"] = strconv.FormatInt(resource.Value(), 10)
 				// For now, only support one GPU type. Break on first match.
 				break
+			} else if resourceKeyString == "aws.amazon.com/neuroncore" && !resource.IsZero() {
+				if err := addNeuronCoresToResourcesIfNotExists(rayStartParams, resource.Value()); err != nil {
```

Comment:
This would be the first hardware accelerator (outside GPUs) that we auto detect in container resource and pass into the `ray start` command as custom resource. I can see the appeal of doing this but it could also mean we end up supporting an ever growing list of custom resources in kuberay.

My biggest question here is whether having to specify the custom resource in `startParams` is a big enough pain point for KubeRay to auto detect custom resources.

Reply:
Thinking about it more, it's probably worth doing for the well-known accelerators at least. Can we generalize the implementation so it's easy to add new accelerators here? cc @ryanaoleary for TPUs

Reply:
Native support for neuron and other accelerators already exists for Ray's VM cluster-launchers.
See discussion in https://github.com/ray-project/ray/issues/44361

This PR would move KubeRay closer to parity with the VM-based solution.

Reply:
I agree that there needs to be some styling here that could generalize to other hardware,
again along the same lines as what exists for Ray's VM support.

Reply:
I have generalized code for adding other accelerator types using a map, let me know if you see any other issues.